src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java

```diff
@@ -37,17 +68,299 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+            }
+            else
+            {
+                Log.getLogger().warn("WorkOrderBuildDecoration.readFromNBT: replace " + sn + " by " + newSN);
+                structureName = newSN.toString();
+
+            }
+        }
+
+        buildingRotation = compound.getInteger(TAG_BUILDING_ROTATION);
+        requested = compound.getBoolean(TAG_IS_REQUESTED);
+        isMirrored = compound.getBoolean(TAG_IS_MIRRORED);
+    }
+
+    /**
+     * Save the Work Order to an NBTTagCompound.
+     *
+     * @param compound NBT tag compound.
+     */
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        BlockPosUtil.writeToNBT(compound, TAG_BUILDING, buildingLocation);
+        if (workOrderName != null)
+        {
+            compound.setString(TAG_WORKORDER_NAME, workOrderName);
+        }
+        compound.setBoolean(TAG_IS_CLEARED, cleared);
+        if (md5 != null)
+        {
+            compound.setString(TAG_SCHEMATIC_MD5, md5);
+        }
+        if (structureName == null)
+        {
+            Log.getLogger().error("WorkOrderBuild.writeToNBT: structureName should not be null!!!");
+        }
+        else
+        {
+            compound.setString(TAG_SCHEMATIC_NAME, structureName);
+        }
+        compound.setInteger(TAG_BUILDING_ROTATION, buildingRotation);
+        compound.setBoolean(TAG_IS_REQUESTED, requested);
+        compound.setBoolean(TAG_IS_MIRRORED, isMirrored);
+    }
+
+    /**
+     * Attempt to fulfill the Work Order.
+     * Override this with an implementation for the Work Order to find a Citizen to perform the job
+     * <p>
+     * finds the first suitable builder for this job.
+     *
+     * @param colony The colony that owns the Work Order.
+     */
+    @Override
+    public void attemptToFulfill(@NotNull final Colony colony)
+    {
+        boolean sendMessage = true;
+        boolean hasBuilder = false;
+
+        for (@NotNull final CitizenData citizen : colony.getCitizens().values())
+        {
+            final JobBuilder job = citizen.getJob(JobBuilder.class);
+
+            if (job == null || citizen.getWorkBuilding() == null)
+            {
+                continue;
+            }
+
+            hasBuilder = true;
+
+            // don't send a message if we have a valid worker that is busy.
+            if (canBuild(citizen))
+            {
+                sendMessage = false;
+            }
+
+            if (!job.hasWorkOrder())
+            {
+                //  A Build WorkOrder may be fulfilled by a Builder as long as any ONE of the following is true:
+                //  - The Builder's Work AbstractBuilding is built
+                //  - OR the WorkOrder is for the Builder's Work AbstractBuilding
+                //  - OR the WorkOrder is for the TownHall
+                if (canBuild(citizen))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)